### PR TITLE
Update logging messages in the VM

### PR DESCRIPTION
**Pull request by Overcastan**
This small PR updates the formatting of the logging messages in the VM to be in the same style with logging messages from Winterfell.

### CODE REVIEW

**Review by bobbinth — COMMENTED** [review]
Looks good! Thank you! I left one question inline.

Also, another question: should we make `tracing-forest` an optional feature and use `tracing-subscriber` by default? (i.e., similar to how Winterfell executable works).

File: assembly/src/assembler/mod.rs [hunk 1]
```diff
@@ -141,7 +141,7 @@ impl Assembler {
     ///
     /// # Errors
     /// Returns an error if the compilation of the specified program fails.
-    #[cfg_attr(feature = "std", instrument("Compile AST", skip_all))]
+    #[cfg_attr(feature = "std", instrument("compile_ast", skip_all))]
```
[the remark this answers is not in the document]
Comment (Overcastan) [reply assembly/src/assembler/mod.rs:144]: To be able to use `instrument` attribute we need to enable `tracing/attributes` feature. For some reason I put this feature in the `std` when implementing logger (I have feeling that I found some docs where was mentioned what `instrument` could work only with `std`, but I can't find it now), so to be able to build project with `—no-default-features` I added these condition. 
Now I see that we can remove this condition, I managed to implement this new version. I will push it together with optional `tracing-forest` implementation.

File: miden/Cargo.toml [hunk 2]
```diff
@@ -56,9 +56,9 @@ serde = {version = "1.0", optional = true }
 serde_derive = {version = "1.0", optional = true }
 serde_json = {version = "1.0", optional = true }
 stdlib = { package = "miden-stdlib", path = "../stdlib", version = "0.8", default-features = false }
-tracing = { version = "0.1", default-features = false, optional = true }
+tracing = { version = "0.1", default-features = false, features = ["attributes"] }
 tracing-subscriber = { version = "0.3", features = ["std", "env-filter"] }
-tracing-forest = { version = "0.1", features = ["ansi", "smallvec"] }
+tracing-forest = { version = "0.1", features = ["ansi", "smallvec"], optional = true }
```
Comment (bobbinth) [inline miden/Cargo.toml:61]: Should these features be enabled only if `executable` feature is enabled?

**Comment by Overcastan** [comment]
I'm wondering should we add a new command to `make` to make enabling `tracing-forest` feature more convenient? Make it similar to the `FEATURES_CONCURRENT_EXEC`, but with `tracing-forest` feature. Something like:
```
FEATURES_LOG_TREE=--features concurrent,executable,tracing-forest

exec-tree:
    cargo build $(PROFILE_OPTIMIZED) $(FEATURES_LOG_TREE)
```

**Comment by bobbinth** [comment]
> I'm wondering should we add a new command to `make` to make enabling `tracing-forest` feature more convenient? Make it similar to the `FEATURES_CONCURRENT_EXEC`, but with `tracing-forest` feature. Something like:

Good idea! I'd probably call it `exec-info`. Also, would we not need to set log level to `INFO` for it to work?

**Comment by Overcastan** [comment]
> Also, would we not need to set log level to `INFO` for it to work?
No, there is no need to change the default log level, we specify it on the execution stage, so the only thing we need to change is a Makefile

**Review by bobbinth — APPROVED** [review]
Looks good! Thank you!

**Pull request merged**